src/transformers/adapters/heads/base.py

```diff
@@ -509,17 +509,25 @@ def _init_head_modules(self):
 
     def get_output_embeddings(self):
         # Only gets the output embeddings for the currently active head
-        if self.active_head in self.heads:
-            head = self.heads[self.active_head]
-            return head.get_output_embeddings()
-        else:
+        embeddings = []
+        for head_name in self._active_heads:
+            if head_name in self.heads:
+                head = self.heads[head_name]
+                output_embeddings = head.get_output_embeddings()
+                embeddings.append(output_embeddings)
+
+        if len(embeddings) == 1:
+            return embeddings[0]
+        elif len(embeddings) == 0 or all([e is None for e in embeddings]):
             return None
+        else:
+            return embeddings
 
     def set_output_embeddings(self, new_embeddings):
         # Only sets the output embeddings for the currently active head
-        if self.active_head in self.heads:
-            head = self.heads[self.active_head]
-            if head.get_output_embeddings() is not None:
+        for head_name in self._active_heads:
+            if head_name in self.heads:
+                head = self.heads[head_name]
```

Comment:
Do I understand it correctly that, in case f multiple heads, with the current implementation `get_output_embeddings()` returns a list of embeddings, and `set_output_embeddings()` only takes one embedding as input that will be set for every head? If so this is kind of unintuitiv

Reply:
Yeah, the implementation is not ideal here. We need `get_output_embeddings()` to return the embeddings of all active heads to make sure we can access them eg. in `freeze_embeddings()`. What would your suggestion be to implement this better?

Reply:
done :)